File: app/src/main/java/com/callmangement/EHR/support/Preference.java
package com.callmangement.EHR.support;

import android.content.Context;
import android.content.SharedPreferences;
import android.content.SharedPreferences.Editor;

import com.callmangement.R;

public class Preference {

    private static Preference preference;
    private static SharedPreferences sharedPreferences;

    public static final String USER_LOGIN_STATUS = "USER_LOGIN_STATUS";
    public static final String USER_ID = "USER_ID";
    public static final String USER_NAME = "USER_NAME";
    public static final String USER_EMAIL = "USER_EMAIL";
    public static final String USER_Mobile = "USER_Mobile";
    public static final String USER_DISTRICT_ID = "USER_DISTRICT_ID";
    public static final String USER_DISTRICT = "USER_DISTRICT";
    public static final String USER_TYPE = "USER_TYPE";
    public static final String USER_TYPE_ID = "USER_TYPE_ID";

    public static Preference getInstance(Context context) {
        if (preference == null) {
            preference = new Preference();
            sharedPreferences = context.getSharedPreferences(context.getResources().getString(R.string.app_name), Context.MODE_PRIVATE);
        }
        return preference;
    }

    public boolean getBoolean(String key) {
        return sharedPreferences.getBoolean(key, false);
    }

    public void putBoolean(String key, boolean value) {
        Editor editor = sharedPreferences.edit();
        editor.putBoolean(key, value);
        editor.apply();
    }

    public String getString(String key) {
        return sharedPreferences.getString(key, "");
    }

    public void putString(String key, String value) {
        Editor editor = sharedPreferences.edit();
        editor.putString(key, value);
        editor.apply();
    }

    public void putInt(String key, int value) {
        Editor editor = sharedPreferences.edit();
        editor.putInt(key, value);
        editor.apply();
    }

    public int getInt(String key) {
        return sharedPreferences.getInt(key, 0);
    }

    public boolean getBoolean(String key, boolean defaultValue) {
        return sharedPreferences.getBoolean(key, defaultValue);
    }

    public void putLong(String key, long value) {
        Editor editor = sharedPreferences.edit();
        editor.putLong(key, value);
        editor.apply();
    }

    public long getLong(String key) {
        return sharedPreferences.getLong(key, 0);
    }

    public void remove(String key) {
        Editor editor = sharedPreferences.edit();
        editor.remove(key);
        editor.apply();
    }

    public void clearPreference() {
        Editor editor = sharedPreferences.edit();
        editor.clear();
        editor.apply();
    }

}

File: app/src/main/java/com/callmangement/ehr/support/Preference.kt
package com.callmangement.ehr.support

import android.content.Context
import android.content.SharedPreferences
import com.callmangement.R

class Preference {
    fun getBoolean(key: String?): Boolean {
        return sharedPreferences!!.getBoolean(key, false)
    }

    fun putBoolean(key: String?, value: Boolean) {
        val editor = sharedPreferences!!.edit()
        editor.putBoolean(key, value)
        editor.apply()
    }

    fun getString(key: String?): String {
        return sharedPreferences!!.getString(key, "")!!
    }

    fun putString(key: String?, value: String?) {
        val editor = sharedPreferences!!.edit()
        editor.putString(key, value)
        editor.apply()
    }

    fun putInt(key: String?, value: Int) {
        val editor = sharedPreferences!!.edit()
        editor.putInt(key, value)
        editor.apply()
    }

    fun getInt(key: String?): Int {
        return sharedPreferences!!.getInt(key, 0)
    }

    fun getBoolean(key: String?, defaultValue: Boolean): Boolean {
        return sharedPreferences!!.getBoolean(key, defaultValue)
    }

    fun putLong(key: String?, value: Long) {
        val editor = sharedPreferences!!.edit()
        editor.putLong(key, value)
        editor.apply()
    }

    fun getLong(key: String?): Long {
        return sharedPreferences!!.getLong(key, 0)
    }

    fun remove(key: String?) {
        val editor = sharedPreferences!!.edit()
        editor.remove(key)
        editor.apply()
    }

    fun clearPreference() {
        val editor = sharedPreferences!!.edit()
        editor.clear()
        editor.apply()
    }

    companion object {
        private var preference: Preference? = null
        private var sharedPreferences: SharedPreferences? = null

        const val USER_LOGIN_STATUS: String = "USER_LOGIN_STATUS"
        const val USER_ID: String = "USER_ID"
        const val USER_NAME: String = "USER_NAME"
        const val USER_EMAIL: String = "USER_EMAIL"
        const val USER_Mobile: String = "USER_Mobile"
        const val USER_DISTRICT_ID: String = "USER_DISTRICT_ID"
        const val USER_DISTRICT: String = "USER_DISTRICT"
        const val USER_TYPE: String = "USER_TYPE"
        const val USER_TYPE_ID: String = "USER_TYPE_ID"

        @JvmStatic
        fun getInstance(context: Context): Preference? {
            if (preference == null) {
                preference = Preference()
                sharedPreferences = context.getSharedPreferences(
                    context.resources.getString(R.string.app_name),
                    Context.MODE_PRIVATE
                )
            }
            return preference
        }
    }
}
